### PR TITLE
Benchmarks: Add Benchmark - Add the source code of cuda kernel launch overhead benchmark.

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Add the cmake project of cuda kernel launch overhead benchmark.

**Major Revision**
- To test kernel launch event time
- To test kernel launch wall time

### CODE REVIEW

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt:1]: Do we have some reference for this? I think it's a general CMakeLists and maybe we can add a reference.

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    {
+        num_steps = std::stoi(value);
+    }
+
+    // Test the kernel launch event time.
+    double event_total_time = test_cuda_kernel_launch_event_time(num_warmups, num_steps);
+    printf("Kernel launch overhead - event time: %3.5f ms \n", event_total_time / num_steps);
+
+    // Sleep for 2 seconds and run the next test.
+    std::this_thread::sleep_for(std::chrono::milliseconds(2000));
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu:106]: Can we add an option for this?

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu [hunk 3]
```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu:1]: Do we need to add UT for this?

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu (hunk 2 above)
Comment (guoshzhao) [reply superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu:106]: Done.

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu (hunk 3 above)
Comment (guoshzhao) [reply superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu:1]: For current design, the source project of microbenchmark is dependency of superbench. If we have indenpendent repo for source projects, we should add UT. But in superbench repo, I think it's not needed.

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt (hunk 1 above)
Comment (guoshzhao) [reply superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt:1]: Yes, I also think it has some common lines, like the ARCHS settings. I referred the cmake file of cutlass. I can extract them into one separated file.

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu (hunk 3 above)
Comment (cp5555) [reply superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/cuda_kernel_launch.cu:1]: OK, Let's wait for Yifan to add Lint.

**Review by cp5555 — APPROVED** [review]

**Review by abuccts — APPROVED** [review]

**Pull request merged**